unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/EmailPromptHUD/EmailPromptHUDController.cs

```diff
@@ -1,34 +1,142 @@
 ﻿using UnityEngine;
-using UnityEngine.UI;
-using TMPro;
 using DCL.Interface;
+using DCL.Helpers;
+using System;
+using System.Collections;
+using DCL.Tutorial;
 
-public class EmailPromptHUDController : MonoBehaviour
+public class EmailPromptHUDController : IHUD
 {
-    public TMP_InputField inputField;
-    public GameObject cryptoUserMessage;
-    public GameObject nonCryptoUserMessage;
-    public Button sendButton;
+    EmailPromptHUDView view;
 
-    void Awake()
+    bool isRendererEnabled = false;
+    bool isTimerComplete = false;
+    bool isTimerCreated = false;
+    bool isEnable = false;
+
+    Coroutine timerRoutine;
+
+    public EmailPromptHUDController()
     {
-        bool hasWallet = UserProfile.GetOwnUserProfile().hasConnectedWeb3;
+        view = UnityEngine.Object.Instantiate(Resources.Load<GameObject>("EmailPromptHUD")).GetComponent<EmailPromptHUDView>();
+        view.name = "_EmailPromptHUD";
+
+        view.OnDismiss += OnDismiss;
+        view.OnSendEmail += OnSendEmail;
 
-        cryptoUserMessage.SetActive(hasWallet);
-        nonCryptoUserMessage.SetActive(!hasWallet);
+        view.gameObject.SetActive(false);
 
-        sendButton.interactable = false;
-        sendButton.onClick.AddListener(SaveEmail);
+        CommonScriptableObjects.rendererState.OnChange += OnRendererStateChanged;
+    }
+
+    public void SetVisibility(bool visible)
+    {
+        if (visible)
+        {
+            Utils.UnlockCursor();
+            view.gameObject.SetActive(true);
+            view.showHideAnimator.Show();
+        }
+        else
+        {
+            view.showHideAnimator.Hide();
+        }
+    }
+
+    public void Dispose()
+    {
+        CommonScriptableObjects.rendererState.OnChange -= OnRendererStateChanged;
+
+        if (view != null)
+        {
+            GameObject.Destroy(view.gameObject);
+        }
+        if (timerRoutine != null)
+        {
+            CoroutineStarter.Stop(timerRoutine);
+        }
+    }
+
+    public void SetEnable(bool enable)
+    {
+        bool wasEnabled = isEnable;
+        isEnable = enable;
+
+        if (enable && !wasEnabled)
+        {
+            if (CommonScriptableObjects.rendererState)
+            {
+                isRendererEnabled = true;
+                CreateTimer();
+            }
+        }
+        else if (!enable && wasEnabled)
+        {
+            if (timerRoutine != null)
+            {
+                CoroutineStarter.Stop(timerRoutine);
+            }
+            isTimerComplete = false;
+            isTimerCreated = false;
+        }
+    }
 
-        inputField.onValueChanged.AddListener(value =>
+    void CreateTimer()
+    {
+        if (!isEnable)
+            return;
+
+        isTimerCreated = true;
+        timerRoutine = CoroutineStarter.Start(WaitForSeconds(60));
```

Comment:
It would be fine to define a constant for this value of 60 seconds 👍

Reply:
You are right. I will create a constant for that. Thanks